javaparser-core/src/main/java/com/github/javaparser/ast/CompilationUnit.java

```diff
@@ -90,23 +90,27 @@
     @InternalProperty
     private Storage storage;
 
+    @InternalProperty
+    private boolean isBad;
+
     public CompilationUnit() {
-        this(null, null, new NodeList<>(), new NodeList<>(), null);
+        this(null, false, null, new NodeList<>(), new NodeList<>(), null);
```

Comment:
for the boolean parameter could be useful to use enums, so that it is more clear what it means